MzML/MzmlMethods.cs

```diff
@@ -828,22 +1073,171 @@ public static void CreateAndWriteMyMzmlWithCalibratedSpectra(IMsDataFile<IMsData
+
+                //spectra naming
+                indexedMzml.indexList.index[0] = new Generated.IndexType()
+                {
+                    name = indexname,
+
+
+                };
+
+                indexname = Generated.IndexTypeName.chromatogram;
+                indexedMzml.indexList.index[0].name = new Generated.IndexTypeName()
+                {
+
+                };
+
+
+                //chroma naming
+                indexedMzml.indexList.index[1] = new Generated.IndexType()
+                {
+                    name = indexname,
+
+                };
+
+
+                indexedMzml.indexList.index[1].name = indexname;
+                {
+
+                };
+
+                int numScans = myMsDataFile.NumSpectra;
+                int numChromas = Int32.Parse(mzML.run.chromatogramList.count);
+
+                //now calculate offsets of each scan and chroma
+
+                //add spectra offsets
+                indexedMzml.indexList.index[0].offset = new Generated.OffsetType[numScans];
+                //add chroma offsets
+                indexedMzml.indexList.index[1].offset = new Generated.OffsetType[numChromas];
+
+
+                var reader = new StreamReader(memStreamWithLineEndings, Encoding.UTF8);
+                memStreamWithLineEndings.Position = 0;
+
+                int i = 0;
+                int a = 1;
+                int index;
+                //spectra offset loop
+                while (i < numScans)
+                {
+                    index = removedBreaks.IndexOf(mzML.run.spectrumList.spectrum[i].id, (a - 1));
+                    if (index != -1)
+                    {
+                        a = index;
+                        indexedMzml.indexList.index[0].offset[i] = new Generated.OffsetType();
+                        indexedMzml.indexList.index[0].offset[i].idRef = mzML.run.spectrumList.spectrum[i].id;
+                        indexedMzml.indexList.index[0].offset[i].Value = a + 3;
+                        i++;
+                    }
+
+                }
+
+                index = removedBreaks.IndexOf("id=\"" + mzML.run.chromatogramList.chromatogram[0].id + "\"", (a - 1));
+                if (index != -1)
+                {
+                    a = index;
+                    indexedMzml.indexList.index[1].offset[0] = new Generated.OffsetType();
+                    indexedMzml.indexList.index[1].offset[0].idRef = mzML.run.chromatogramList.chromatogram[0].id;
+                    indexedMzml.indexList.index[1].offset[0].Value = a + 3;
                 }
+
+                indexedMzml.indexListOffset = indexListOffset - 32;
+                // 4. Write indexed mzML object to string in memory
+
+
+                string chksum = "";
+
+                //Write to file
+
+                TextWriter writer = new StreamWriter(outputFile);
+
+                writer.NewLine = "\n";
+                writer.Close();
+                var stream = File.OpenRead(outputFile);
```

Comment:
I believe this computes checksum of <indexedmzml> ... </indexededmzml>

Instead of  from <indexededmzml> ... <checksum>